addons/mail/models/mail_channel.py

```diff
@@ -312,6 +331,134 @@ def _can_invite(self, partner_id):
+            'rtc_ringing_partner_id': False
+        })
+        notifications = []
+        for partner in invited_members.mapped('partner_id'):
+            notifications.append([
+                (self._cr.dbname, 'res.partner', partner.id),
+                {
+                    'type': 'rtc_invitation',
+                    'payload': {
+                        'remove': True,
+                        'channelId': self.id,
+                    },
+                },
+            ])
+        self.env['bus.bus'].sendmany(notifications)
+
+    def _invite_members_to_rtc(self):
+        """ Sends invitations to join the RTC call to all connected members of the thread who are not already invited.
+        """
+        self.ensure_one()
+        current_partner = self.env.user.partner_id
+        current_sessions_partners = self.rtc_sessions.mapped('partner_id')
+        notifications = []
+        for channel_partner in self.channel_last_seen_partner_ids:
+            if channel_partner.partner_id == current_partner:
+                continue
+            if channel_partner.partner_id in current_sessions_partners:
+                continue
+            if channel_partner.rtc_ringing_partner_id:
+                continue
+            channel_partner.rtc_ringing_partner_id = current_partner.id
+            notifications.append([
+                (self._cr.dbname, 'res.partner', channel_partner.partner_id.id),
+                {
+                    'type': 'rtc_invitation',
+                    'payload': {
+                        'channelId': self.id,
+                        'partner': {
+                            'id': current_partner.id,
+                            'name': current_partner.name,
+                        },
+                    },
+                },
+            ])
+        notification = _(
+            '%s started a live conference',
+            current_partner.name)
+        self.message_post(body=notification, message_type="notification")
+        self.env['bus.bus'].sendmany(notifications)
+
+    def _join_call(self):
+        session_data, session_id = self._update_call_participation(True)
+        ice_servers = self.env['mail.ice.server']._get_ice_servers()
+        if len(self.rtc_sessions) == 1 and self.channel_type in ['chat', 'group']:
+            self._invite_members_to_rtc()
+        return {
+            'rtcSessions': session_data,
+            'iceServers': ice_servers or False,
+            'sessionId': session_id,
+        }
+
+    def _leave_call(self, session_id=None):
+        self._update_call_participation(False, session_id=session_id)
+
+    def _update_call_participation(self, join_call, session_id=None):
+        """ Updates the call participation of the current partner and notifies members of
+            the channel if necessary.
+
+            :param join_call : boolean true if joining the call, false if leaving.
+            :param session_id : int to be provided if leaving the call
```

Comment:
```suggestion
            :param bool join_call : true if joining the call, false if leaving.
            :param int session_id : to be provided if leaving the call
```
this allows editors to get the type information even without manifest typing.

Also `join_call` is a bit odd as a parameter name, maybe `joining` would be clearer?

And it would benefit from requiring keyword arguments, the callsites are... rather obscure. Either that or use an `enum` with suitably named members (e.g. `Joining` and `Leaving`) but that seems a bit brutal.